ledger/participant-integration-api/src/main/scala/platform/indexer/ha/PollingChecker.scala

```diff
@@ -61,5 +64,10 @@ class PollingChecker(
     }
   }
 
-  def close(): Unit = timer.cancel()
+  private var closed: Boolean = false
+
+  def close(): Unit = synchronized {
+    closed = true
```

Comment:
As I understand cancel kills all waiting timer tasks, but not the tasks that are already running. 
There might already be a check function ready to run when this line is executed. This will lead to the Exception being thrown. Am I missing something?

Reply:
critical sections demarcated by synchronized are mutually exclusive, and waiting for each other to enter.
check is also demarcated by synchronized

Reply:
check is demarcated by synchronize but the TimerTask.run() method is not. In fact run() method can be waiting to enter the critical section in Try(check()) as closed=true is being written.
The safest way to cancel a timer is to write a variable then leave inside the check() method first thing.

Reply:
not sure what are you suggesting exactly, bringing this offline?

Reply:
We have gone over this offline and the challenge here is that the check function is called from 2 places: from the timer and from HaCoordinator. It is desirable to throw in the latter case, while timer is anyway enclosed in a Try-do-nothing clause.

Reply:
we leave it as discussed offline